### PR TITLE
Virtual Turtor Backend Merge

### PR DESCRIPTION
**Pull request by SlovenlyTF**
## Description

[Web Create] As a content creator, I want to be able to put information on the new section of the new course I'm creating.
[Web Create] As a content creator, I want to be able to put general information on the new courses I'm creating.
This PBI is almost completely done, but you still can't upload a cover image.
[Web Create] As a content creator, I want to be able to add a section on the new course I'm creating.
[Web Edit] As a content creator, I want to be able to edit the sections of the courses I haven't published yet.
[Web Edit] As a content creator, I want to be able to edit the general info of the courses I haven't published yet

## Changes

* Created routes for Section
  * Can now create a section
  * Can now delete a section
  * Can now update data in a section
* Created routes for courses
  * Can now create a course
  * Can now delete a course
  * Can now update data in a course
* Created lectureRoutes
  * Can now create a lecture
  * Can now delete a lecture
  * Can now update data in a lecture
* Created routes for exercises
  * Can now create an exercises
  * Can now update the data in an exercises

## Checklist

- [x] Code has been tested locally and passes all relevant tests.
- [x] Documentation has been updated to reflect the changes, if applicable.
- [x] Code follows the established coding style and guidelines of the project.
- [x] All new and existing tests related to the changes have passed.
- [x] Any necessary dependencies or new packages have been properly documented.
- [x] Pull request title and description are clear and descriptive.
- [x] Reviewers have been assigned to the pull request.
- [x] Any potential security implications have been considered and addressed.
- [x] Performance impact of the changes has been evaluated, if relevant.

### CODE REVIEW

**Review by Adachigaming — APPROVED** [review]
I only checked to see if there were any conflicts with the Onboarding web functionalities. Looks good as far as I'm concerned.

File: __tests__/fixtures/fakeSection.js
Comment (Muad3) [inline __tests__/fixtures/fakeSection.js:1]: er der en grund til at i laver fake course og sections, istedet for at bruge dem i databasen?

Comment (SlovenlyTF) [reply __tests__/fixtures/fakeSection.js:1]: Ja, det er fordi vi bruger den til at teste, så for ikke at lave en masse sections og ting i databasen, så bruger vi "fake" sections og ting.

**Review by Muad3 — CHANGES_REQUESTED** [review]
mangler måske nogle flere tests

**Comment by SlovenlyTF** [comment]
Har lavet nogle tests for lectures og exercises👍
Håber det er okay nu :D

**Review by Muad3 — APPROVED** [review]
synes det fint

**Pull request merged**